New package: pdfarranger-1.7.1_1

hervyqa opened the pull request:
<!-- Mark items with [x] where applicable -->

#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

<!--
If GitHub CI cannot be used to validate the build result (for example, if the
build is likely to take several hours), make sure to
[skip CI](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration).
When skipping CI, uncomment and fill out the following section.
Note: for builds that are likely to complete in less than 2 hours, it is not
acceptable to skip CI.
-->

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (x86_64-musl)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl

hervyqa commented:
```
=> pdfarranger-1.7.1_1: running do_check ...
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
creating pdfarranger.egg-info
writing pdfarranger.egg-info/PKG-INFO
writing dependency_links to pdfarranger.egg-info/dependency_links.txt
writing entry points to pdfarranger.egg-info/entry_points.txt
writing requirements to pdfarranger.egg-info/requires.txt
writing top-level names to pdfarranger.egg-info/top_level.txt
writing manifest file 'pdfarranger.egg-info/SOURCES.txt'
reading manifest file 'pdfarranger.egg-info/SOURCES.txt'
adding license file 'COPYING'
writing manifest file 'pdfarranger.egg-info/SOURCES.txt'
running build_ext
Unable to init server: Could not connect: Connection refused
Unable to init server: Could not connect: Connection refused
/builddir/pdfarranger-1.7.1/pdfarranger/splitter.py:17: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '3.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk

(setup.py:3814): Gtk-CRITICAL **: 18:11:15.810: gtk_clipboard_get_for_display: assertion 'display != NULL' failed

(setup.py:3814): Gtk-CRITICAL **: 18:11:15.810: gtk_clipboard_get_for_display: assertion 'display != NULL' failed
Unable to init server: Could not connect: Connection refused

(setup.py:3814): Gtk-WARNING **: 18:11:15.811: cannot open display: 
=> ERROR: pdfarranger-1.7.1_1: do_check: 'python3 setup.py ${make_check_target} ${make_check_args}' exited with 1
=> ERROR:   in do_check() at common/build-style/python3-module.sh:40
Error: Process completed with exit code 1.
```

Run test failed. any ideas?

paper42 commented:
It's saying that it needs a graphical session, you can fix it with something like:
```sh
do_check() {
	xvfb-run python3 setup.py test
}
```
Package xvfb-run needs to be added to checkdepends to make this work.

paper42 commented on srcpkgs/pdfarranger/template:
```diff
@@ -0,0 +1,23 @@
+version=1.7.1
+revision=1
+build_style=python3-module
+hostmakedepends="
```
start listing the packages on this line, no need for a linebreak

hervyqa replied:
ok. I have deleted.

paper42 commented on srcpkgs/pdfarranger/template:
```diff
@@ -0,0 +1,23 @@
+build_style=python3-module
+hostmakedepends="
+ ghostscript intltool python3-setuptools python3-distutils-extra python3-pikepdf
+ python3-pip python3-xlib python3-wheel"
```
does it really need pip?

hervyqa replied:
seems unnecessary. previously also worked without python3-pip (local compile). it's just that i see it in debian dependencies.

paper42 commented on srcpkgs/pdfarranger/template:
```diff
@@ -0,0 +1,23 @@
+depends="
+ gtk+3 python3-cairo python3-gobject python3-pikepdf python3-dateutil
+ ghostscript python3-setuptools python3-xlib poppler-glib img2pdf"
+checkdepends="python3-wheel python3-pip $depends"
```
Can the checks work without pip? With pip they are useless and should be disabled.

hervyqa replied:
oke. i have deleted.

hervyqa commented:
> It's saying that it needs a graphical session, you can fix it with something like:
> 
> ```shell
> do_check() {
> 	xvfb-run python3 setup.py test
> }
> ```
> 
> Package xvfb-run needs to be added to checkdepends to make this work.

thank you @paper42, it worked 😉

rivanfebrian123 commented:
Hi @hervyqa! I appreciate your work, but isn't this a duplicate of #30338?

paper42 commented:
If @noarchwastaken is interested in this continuing the work on pdfarranger, I will to close this PR as a duplicate of #30338, but @hervyqa, please create a review on that PR, because some parts of this one are better and should be moved there.

If they are not interested in it anymore, we can continue here and pull in some of their changes.

@rivanfebrian123 thanks for noticing this

hervyqa commented:
when compared to PR #30338 there are a few points that I corrected.

1. #30338 some checks didn't work
2. I've already added dependencies before building the packages.
3. The dependency for running package checks is included as well, while #30338 is not.

by the way, thanks @paper42 and @rivanfebrian123 for the information.

noarchwastaken commented:
> If @noarchwastaken is interested in this continuing the work on pdfarranger, I will to close this PR as a duplicate of #30338, but @hervyqa, please create a review on that PR, because some parts of this one are better and should be moved there.
> 
> If they are not interested in it anymore, we can continue here and pull in some of their changes.
> 
> @rivanfebrian123 thanks for noticing this

I'll go with this PR since it has the above fixes.

About the `xvfb-run` dependency, I can't find any checks included with `pdfarranger`, so what does `setup.py test` actually do here?

hervyqa commented:
> > If @noarchwastaken is interested in this continuing the work on pdfarranger, I will to close this PR as a duplicate of #30338, but @hervyqa, please create a review on that PR, because some parts of this one are better and should be moved there.
> > If they are not interested in it anymore, we can continue here and pull in some of their changes.
> > @rivanfebrian123 thanks for noticing this
> 
> I'll go with this PR since it has the above fixes.
> 
> About the `xvfb-run` dependency, I can't find any checks included with `pdfarranger`, so what does `setup.py test` actually do here?

Needs a graphical session. https://github.com/void-linux/void-packages/pull/32301#issuecomment-891248272

```
running build_ext
Unable to init server: Could not connect: Connection refused
Unable to init server: Could not connect: Connection refused
/builddir/pdfarranger-1.7.1/pdfarranger/splitter.py:17: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '3.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk

(setup.py:3814): Gtk-CRITICAL **: 18:11:15.810: gtk_clipboard_get_for_display: assertion 'display != NULL' failed

(setup.py:3814): Gtk-CRITICAL **: 18:11:15.810: gtk_clipboard_get_for_display: assertion 'display != NULL' failed
Unable to init server: Could not connect: Connection refused

(setup.py:3814): Gtk-WARNING **: 18:11:15.811: cannot open display: 
=> ERROR: pdfarranger-1.7.1_1: do_check: 'python3 setup.py ${make_check_target} ${make_check_args}' exited with 1
=> ERROR:   in do_check() at common/build-style/python3-module.sh:40
Error: Process completed with exit code 1.
```

paper42 commented:
> About the `xvfb-run` dependency, I can't find any checks included with `pdfarranger`, so what does `setup.py test` actually do here?

I assume it runs tests/test.py

hervyqa commented:
hi @paper42, do you have any suggestions for improvement?

hervyqa commented:
new version #34216
